wgpu-hal/src/gles/device.rs

```diff
@@ -676,14 +724,16 @@ impl crate::Device<super::Api> for super::Device {
         })
     }
     unsafe fn destroy_texture(&self, texture: super::Texture) {
-        let gl = &self.shared.context.lock();
-        match texture.inner {
-            super::TextureInner::Renderbuffer { raw, .. } => {
-                gl.delete_renderbuffer(raw);
-            }
-            super::TextureInner::DefaultRenderbuffer => {}
-            super::TextureInner::Texture { raw, .. } => {
-                gl.delete_texture(raw);
+        if texture.drop_guard.is_none() {
```

Comment:
Where do we actually call the drop guard?

Reply:
Isnt the drop guard's Drop implementation called at the end of the scope since this function takes ownership of the hal texture?

Reply:
Ah yes, could we just call that out either with a explicit call to `drop(texture)` or a comment?